agent/discovery/cloud.go

```diff
@@ -24,23 +26,23 @@ func (d CloudDiscovery) GetId() string {
 	return d.id
 }
 
-func (d CloudDiscovery) Discover() error {
+func (d CloudDiscovery) Discover() (string, error) {
 	cloudData, err := cloud.NewCloudInstance()
 	if err != nil {
-		return err
+		return fmt.Errorf("%v", err).Error(), err
 	}
 
 	err = cloudData.Store(d.discovery.client)
 	if err != nil {
-		return err
+		return fmt.Errorf("%v", err).Error(), err
 	}
 
 	err = storeCloudMetadata(d.discovery.client, cloudData.Provider)
 	if err != nil {
-		return err
+		return fmt.Errorf("%v", err).Error(), err
 	}
 
-	return nil
+	return "Found " + cloudData.Provider + " cloud provider", nil
```

Comment:
We should use the `fmt.Sprintf` options here to format the string

Reply:
And to follow the same convention as the other discovery systems, I would rename to `fmt.Sprintf("Cloud provider %s discovered", cloudData.Provider)`

Reply:
Changed!